### PR TITLE
Replaced Iterators with foreachloops

### PR DESCRIPTION
**Pull request by RoiEXLab**
This is part 1 of the Iterator Replacement...
Only 1000 "iterator entries" left ^^

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/773)

### CODE REVIEW

File: src/games/strategy/engine/chat/PlayerChatRenderer.java [hunk 1]
```diff
@@ -47,13 +46,12 @@ public Component getListCellRendererComponent(final JList<?> list, final Object
       final Set<String> players = m_playerMap.get(value.toString());
       if (players != null && !players.isEmpty()) {
         sb.append(" (");
-        final Iterator<String> iter = players.iterator();
-        while (iter.hasNext()) {
-          sb.append(iter.next());
-          if (iter.hasNext()) {
-            sb.append(", ");
-          }
+        for(String player : players) {
```
Comment (DanVanAtta) [inline src/games/strategy/engine/chat/PlayerChatRenderer.java]: Use Guava Joiner instead, it'll be something like:
`Joiner.on(",").join(players)`

File: src/games/strategy/engine/data/export/DefaultAttachmentExporter.java [hunk 2]
```diff
@@ -138,12 +134,12 @@ private String printStringArrayOption(final Field field, final String option, fi
       if (valueArray == null) {
         return "";
       }
-      final Iterator<String> values = Arrays.asList(valueArray).iterator();
-      String value = values.next();
-      while (values.hasNext()) {
-        value = value + ":" + values.next();
+      StringBuilder valueBuilder = new StringBuilder();
+      for(String val : valueArray) {
+        valueBuilder.append(":").append(val);
```
Comment (DanVanAtta) [inline src/games/strategy/engine/data/export/DefaultAttachmentExporter.java]: Same, can this be replaced with a Joiner? Ditto for the any other StringBuilders that we use in a loop.

**Comment by DanVanAtta** [comment]
Looks pretty alright : )

See if you can use a "Joiner" to simplify/replace some of the loops

**Comment by RoiEXLab** [comment]
Seems to be a good idea...

**Comment by RoiEXLab** [comment]
Done

**Pull request merged**